Myc 11 fix data implementation

GabrielKidane96 opened the pull request:
Internal review for Sphereon prior to moving this to the lto repository.
Currently doing some cleanup, will move from draft once finished.

Commit:
WIP: Fix Data Implementation
Update balance retrieval
Add transaction fees

Commit:
WIP: MYC-11
Added missing transaction types
Updated rosetta test
Added Operation types

Commit:
WIP: MYC-11
Updated balance retrieval
Add reward transactions
Fix block url

Commit:
MYC-11 Fix Data Implementation
Added Data Transaction
Added Script Transaction
Updated Sponsor transaction logic
Added persistant storage for sponsors

nklomp commented on src/secrets/secrets.ts:
```diff
@@ -6,8 +6,8 @@ if (fs.existsSync(".env")) {
 } else {
-    logger.debug("Using .env.example file to supply config environment variables");
-    dotenv.config({ path: ".env.example" });  // you can delete this after you create your own .env file!
+    logger.debug("Using .env file to supply config environment variables");
```
Although there should be an explicit check for the example file, I don't believe the change to .env is correct here, given the expression in the if already checks for the .env file

GabrielKidane96 replied:
Not clear what I should be doing here.

nklomp replied:
Well is there a specific reason why you changed the logic here?

GabrielKidane96 replied:
Don't recall making these changes, have reverted them.

Commit:
MYC-11 Fix data implementation
Remove unused cli configuration
Update storage location
update balance retrieval

Commit:
MYC-11 Fix Data Implementation
Update directory naming

nklomp commented on src/types/Transaction.ts:
```diff
@@ -69,23 +87,49 @@ export class Transaction {
-            case 2: {
-                return this.getPaymentOperations();
-            }
+            // case 2: {
```
Any specific reason this is commented right now? If it is not available in LTO, simply delete it together with the getPaymentOperations function

GabrielKidane96 replied:
removed

nklomp commented on src/types/Transaction.ts:
```diff
@@ -120,41 +164,161 @@ export class Transaction {
     }
 
+    private async getSponsor(address: string): Promise<string> {
+        const sponsors = await storage.getItem('sponsors') as Sponsorship[] || [];
```
Probably it makes sense to store the start block and to take current block whilst calling this function into account.

Let's say rosetta is syncing and it reached a sponsor start transaction. If I then call into methods for pre-sponsor transactions than the logic probably goes wrong, because the state currently acts as a boolean without context

GabrielKidane96 replied:
Updated, I would still want to test this out though

GabrielKidane96 replied:
Updated and tested

nklomp commented on src/server.ts:
```diff
@@ -12,15 +13,40 @@ if (!app.get("prod")) {
     );
     console.log("  Press CTRL-C to stop\n");
+    await setStorage();
+    console.log('Storage now live')
```
Maybe make the log message a bit more clear.

GabrielKidane96 replied:
Updated

nklomp reviewed (changes requested):
See remarks

Commit:
MYc-11 Fix Data Implementation
Remove unused transaction type
Add block height to sponsor storage

Commit:
MYC-11 Fix Data Implementation
Updated log statement for state storage

Commit:
MYC-11 Fixed height retrieval

nklomp reviewed (changes requested):
There still is a logic change in the .env(.example) handling in this PR. I think the change shouldn't be there, and probably is a result from your initial setup "strugles"

Commit:
MYC-11 remove secrets changes
